addons/account/models/account_move.py

```diff
@@ -3835,7 +3835,7 @@ def _get_violated_lock_dates(self, invoice_date, has_tax):
         :return: a list of tuples containing the lock dates affecting this move, ordered chronologically.
         """
         locks = []
-        user_lock_date = self.company_id._get_user_fiscal_lock_date()
+        user_lock_date = self.company_id and self.company_id._get_user_fiscal_lock_date()
```

Comment:
You cannot just add a check that prevents traceback. You have to justify that new version doesn't break logic behind the code you change. 
Sometimes, a solution for traceback could be raising `UserError`/`ValidationError`.